### ISSUE TITLE
Does ShellCheck not fully support `env -S`?

### ISSUE THREAD

**Issue by scalisi** [koalaman/shellcheck#3123]
On [2021-09-18](https://github.com/koalaman/shellcheck/issues/833#issuecomment-922361300), @koalaman wrote: 
> ShellCheck has started supporting `env -S`

I tested ShellCheck's support for `env -S` at [ShellCheck.net](https://www.shellcheck.net/) by entering the following code, which produced an [SC2164](https://www.shellcheck.net/wiki/SC2164) warning as expected:

```bash
#!/usr/bin/env -S bash
cd foo
```

As expected, the following code **does not** produce an SC2164 warning:

```bash
#!/usr/bin/env -S bash -e
cd foo
```

Unexpectedly, the following code also **does not** produce an SC2164 warning:

```bash
#!/usr/bin/env -S bash -o foobared
cd foo
```

It seems that the letter "e" in `-o foobared` (as well as any lowercase "e" after `env -S bash`) disables the SC2164 warning. When the letter "e" is removed from `-o foobared`, the code **does** produce an SC2164 warning:

```bash
#!/usr/bin/env -S bash -o foobard
cd foo
```

Does ShellCheck not fully support `env -S`?
